components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/context/model/ApplicationEntity.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.core.context.model;
+
+/**
+ * Entity class for Application.
+ * This class holds the application entity details for a given flow.
+ */
+public class ApplicationEntity implements InitiatingEntity {
+
+    private final String applicationId;
+    private final String applicationName;
+    private final String clientId;
+    private final String issuer;
+    private final String realm;
```

Comment:
Not clear on what these attributes means, specially the realm.. Is that a generalized analogy to indicate tenant or org?

Also `clientId`/'issuer' are these related to keep the protocol level client's reference in oauth & saml? If so we shouldn't keep the protocol specific attributes in this level.

Reply:
Fixed